### PR TITLE
Add Purpose mutations to base AuthSerializerMutation

### PR DESCRIPTION
**Pull request by nicobav**
Purpose mutations (create and update) are serializer mutations
those needs to be based on AuthSerializerMutation to get permission to
function properly.

Adds permission tests to Purpose mutations.

### CODE REVIEW

**Review by jmp — COMMENTED** [review]
Looks good! I left a couple of minor comments.

File: api/graphql/tests/test_purposes.py [hunk 1]
```diff
@@ -80,3 +83,45 @@ def test_creating_purpose(self):
...
+            'createPurpose(input: {name: "Created purpose"}) {\n'
+            "purpose {\n"
+            "name\n"
+            "}\n"
+            "errors {\n"
+            "messages\n"
+            "field\n"
+            "}\n"
+            "}\n"
+            "}\n"
```
[the remark this answers is not in the document]
Comment (nicobav) [reply api/graphql/tests/test_purposes.py:100]: I agree and I was about to do that but was lazy and just copy/pasted the query strings from the existing tests.

**Comment by nicobav** [comment]
I made the test look nicer :innocent:

**Review by jmp — APPROVED** [review]
Nice! 🚀

**Pull request merged**